com.unity.netcode.gameobjects/Runtime/Spawning/NetworkSpawnManager.cs

```diff
@@ -314,40 +314,52 @@ internal bool HasPrefab(NetworkObject.SceneObject sceneObject)
+            var globalObjectIdHash = sceneObject.Header.Hash;
+            var position = sceneObject.Header.HasTransform ? sceneObject.Transform.Position : default;
+            var rotation = sceneObject.Header.HasTransform ? sceneObject.Transform.Rotation : default;
+            var isReparented = sceneObject.Header.IsReparented;
+            var parentNetworkId = sceneObject.Header.HasParent ? sceneObject.ParentObjectId : default;
+            var worldPositionStays = sceneObject.Header.HasParent ? sceneObject.WorldPositionStays : true;
 
-            if (parentNetworkId != null && !isReparented)
+            if (sceneObject.Header.HasParent)
             {
-                if (SpawnedObjects.TryGetValue(parentNetworkId.Value, out NetworkObject networkObject))
+                if (SpawnedObjects.TryGetValue(sceneObject.ParentObjectId, out NetworkObject networkObject))
                 {
                     parentNetworkObject = networkObject;
                 }
-                else
+                else if (isReparented) // TODO: Not sure we need this warning any longer
                 {
-                    if (NetworkLog.CurrentLogLevel <= LogLevel.Normal)
+                    if (NetworkLog.CurrentLogLevel <= LogLevel.Developer)
                     {
                         NetworkLog.LogWarning("Cannot find parent. Parent objects always have to be spawned and replicated BEFORE the child");
                     }
                 }
             }
 
-            if (!NetworkManager.NetworkConfig.EnableSceneManagement || !isSceneObject)
+            if (!NetworkManager.NetworkConfig.EnableSceneManagement || !sceneObject.Header.IsSceneObject)
             {
                 // If the prefab hash has a registered INetworkPrefabInstanceHandler derived class
                 if (NetworkManager.PrefabHandler.ContainsHandler(globalObjectIdHash))
                 {
-                    // Let the handler spawn the NetworkObject
-                    var networkObject = NetworkManager.PrefabHandler.HandleNetworkPrefabSpawn(globalObjectIdHash, ownerClientId, position.GetValueOrDefault(Vector3.zero), rotation.GetValueOrDefault(Quaternion.identity));
 
+                    // Let the handler spawn the NetworkObject
+                    var networkObject = NetworkManager.PrefabHandler.HandleNetworkPrefabSpawn(globalObjectIdHash, sceneObject.Header.OwnerClientId, position, rotation);
                     networkObject.NetworkManagerOwner = NetworkManager;
 
+                    // Go ahead and set network parenting properties
+                    networkObject.SetNetworkParenting(isReparented, parentNetworkId, worldPositionStays);
                     if (parentNetworkObject != null)
                     {
-                        networkObject.transform.SetParent(parentNetworkObject.transform, true);
+                        networkObject.TrySetParent(parentNetworkObject.transform, worldPositionStays);
```

Comment:
What's the rationale for not setting position & rotation if there's a handler?

Reply:
Let me review this area.

Reply:
So, the position and rotation are sent to the prefab handler and the prefab handler determines whether it will apply those values or not (up to the user).